### ISSUE TITLE
PHP 8.4 compatibility

### ISSUE THREAD

**Issue by mikkamp** [googleapis/google-api-php-client#2636]
We are bundling this library within our extension through composer and we've been working on PHP 8.4 compatibility and I ran into several deprecation messages about using an explicit nullable type.

```
Deprecated: Vendor\Google\Client::authorize(): Implicitly marking parameter $http as nullable is deprecated, the explicit nullable type must be used instead in vendor/google/apiclient/src/Client.php on line 451
```

```diff
-    public function authorize(ClientInterface $http = null)
+    public function authorize(?ClientInterface $http = null)
```

```
Deprecated: Vendor\Google\Client::fetchAccessTokenWithAssertion(): Implicitly marking parameter $authHttp as nullable is deprecated, the explicit nullable type must be used instead in vendor/google/apiclient/src/Client.php on line 318
```

```diff
-        public function fetchAccessTokenWithAssertion(ClientInterface $authHttp = null)
+        public function fetchAccessTokenWithAssertion(?ClientInterface $authHttp = null)
```

```
Implicitly marking parameter $previous as nullable is deprecated, the explicit nullable type must be used instead
vendor/google/apiclient/src/Service/Exception.php:39
```

```diff
    public function __construct(
        $message,
        $code = 0,
-        Exception $previous = null,
+        ?Exception $previous = null,
        $errors = []
    ) {
```

#### Environment details

  - OS: multiple
  - PHP version: 8.4
  - Package name and version: google/apiclient

**Comment by cococaline** [comment]
i face the same issue on php 8.4

**Comment by mikkamp** [comment]
Seems there is already a PR available that would resolve these issues: #2633
